rst_files/generic_programming.rst

```diff
@@ -426,7 +426,7 @@ Numbers and Algebraic Structures
 
 Define two binary functions,  :math:`+` and :math:`\cdot`, called addition and multiplication -- although the operators can be applied to data structures much more abstract than a ``Real``
 
-In mathematics, a `ring <https://en.wikipedia.org/wiki/ring_\(mathematics\>`_ is a set with associated additive and multiplicative operators where
+In mathematics, a `ring <https://goo.gl/unRJRB>`_ is a set with associated additive and multiplicative operators where
```

Comment:
I don't really like this approach.  I would rather people be able to copy/see the URL, assuming that the escaping of RST is fixed.  But if there is no other way, then so be it.

Reply:
this will work. This is my last change. 

The other formats didn't work properly.

Reply:
Ok then it is a different issue for sphinx/jupinx from what we discovered before. Because, external links for websites work all the time, except for web addresses which have "("  ")", I think.